code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -1122,6 +1124,8 @@
 	else
 		var/selection = tgui_input_list(src, "Select an icon! [triesleft ? "You have [triesleft] more chance\s." : "This is your last try."]", "Robot Icon", module_sprites)
 		sprite_datum = selection
+		if(!istype(src,/mob/living/silicon/robot/drone))
+			robot_species = selection

```

Comment:
I'd suggest going for 'sprite_datum.name' instead of 'selection' here

Reply:
Why, the selection is the name of the sprite you picked as it appears in the list.

Reply:
Selection is actually sprite itself, and it might be byond jank but it automatically applies name var to the list/converts it when doing "[selection]" *despite* the fact that name is not a native datum var and its weird like that and this is mostly pedantics.